scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala

```diff
@@ -50,16 +53,48 @@ class SpelExpressionSuggester(
         validationContext
       )
 
-    lazy val newExpression: Expression = truncateExpressionByCaretPosition2d(expression, caretPosition2d)
+    lazy val newExpressionForSpELVariable: Option[Expression] =
+      truncateExpressionToCorrespondingSpELVariable(expression, caretPosition2d)
 
-    lazy val futureSuggestionsAfterTruncatingExpressionByCaretPositionOption = expressionSuggestionsAux(
-      newExpression,
-      caretPosition2d.normalizedCaretPosition(newExpression.expression),
-      validationContext
-    )
+    lazy val futureSuggestionsAfterTruncatingExpressionToCorrespondingSpELVariableOption =
+      newExpressionForSpELVariable.flatMap(e =>
+        expressionSuggestionsAux(
+          e,
+          e.expression.length,
+          validationContext
+        )
+      )
 
-    val firstNonEmptySuggestionFuture =
-      futureSuggestionsOption orElse futureSuggestionsAfterTruncatingExpressionByCaretPositionOption getOrElse successfulNil
+    def processOptions[A](options: LazyList[Option[Future[Iterable[A]]]]): Future[Iterable[A]] = {
```

Comment:
The name `processOptions` is pretty generic. How about something more concrete like 'firstNonEmptyFuture'? And maybe extract this as a private method to make this `expressionSuggestions` method more readable?